### PR TITLE
Lodash: Only use where needed - isFunction

### PR DESCRIPTION
**Pull request by nikolasrieble**
## Description
No need to use lodash for isFunction. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/util/isFunction.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function isFunction(value: unknown): value is Function {
+  return typeof value === 'function';
```
Comment (PavelVanecek) [inline src/util/isFunction.ts:2]: So I looked into this recently and discovered that lodash is doing something way more complicated than`typeof`: https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L3063

Are we okay with replacing it with `typeof` check? I saw some Safari bugs mentioned in there saying that `typeof () => {}` returns `Object`. But I didn't go in the detail.

Comment (HHongSeungWoo) [reply src/util/isFunction.ts:2]: https://github.com/HHongSeungWoo/recharts/blob/0c1aa93cb4ea2de06a31b8efce6a1a27f446af7d/src/util/DataUtils.ts#L19 
How about doing it this way?

**Comment by ckifer** [comment]
@nikolasrieble what would you  like to do with this and the other PR?

File: src/util/isFunction.ts (hunk 1 above)
Comment (nikolasrieble) [reply src/util/isFunction.ts:2]: As https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L3063 indeed seems to handle more edge cases, I believe it is safest to just still use lodash.isfunction

**Pull request closed without merging**